src/pages/home/report/ReportActionsList/DeviceEventListenerRef.ts

```diff
@@ -0,0 +1,30 @@
+import {DeviceEventEmitter, EmitterSubscription} from 'react-native';
+
+export default class DeviceEventListenerRef {
+    private typePrefix: string;
+
+    private subscription: EmitterSubscription | null = null;
+
+    constructor(typePrefix: string) {
+        this.typePrefix = typePrefix;
+    }
+
+    static use(typePrefix: string): DeviceEventListenerRef {
+        return new DeviceEventListenerRef(typePrefix);
```

Comment:
I thought of an API that would directly reflect the previous usage. I meant a 100% direct refactoring without any change in the behavior.

We would use `useRef(null);` here and save it in the class private field. Then we'd use `DeviceEventListenerRef.use()` in the place we called `useRef(null);` before.

I'm **not** saying that what you did doesn't work, as it actually feels clearer.

We could go two ways:

- We could make the implementation directly reflect the old logic (my original intention)
- ...or, we could stay with your way, which is somewhat clearer, but:
  - we'd need to rename some things, as `DeviceEventListenerRef` and `use` names were directly related with the hook nature of my suggestion
  - we'd need to be 100% sure that this doesn't break anything, as the use-site of this logic is quite convoluted